examples/multipro/protocol.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"bufio"
+	p2p "github.com/libp2p/go-libp2p/examples/multipro/pb"
+	protobufCodec "github.com/multiformats/go-multicodec/protobuf"
+	inet "gx/ipfs/QmbD5yKbXahNvoMqzeuNyKQA9vAs9fUvJg2GXeWU1fVqY5/go-libp2p-net"
+	"log"
+	"time"
+	//host "gx/ipfs/QmRS46AyqtpJBsf1zmQdeizSDEzo1qkWR7rdEuPFAv8237/go-libp2p-host"
+	//"bytes"
+	"github.com/gogo/protobuf/proto"
+)
+
+// node version
+const clientVersion = "go-p2p-node/0.0.1"
+
+// helper method - writes a protobuf go data object to a network stream
+// data - reference of protobuf go data object to send (not the object itself)
+// s - network stream to write the data to
+func sendProtoMessage(data proto.Message, s inet.Stream) bool {
+	writer := bufio.NewWriter(s)
+	enc := protobufCodec.Multicodec(nil).Encoder(writer)
+	err := enc.Encode(data)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
log.Fatal will halt the program immediately, we would never get to return false

Reply:
good catch - got rid of all log.Fatal in latest check-in